nisystemlink/clients/result/_result_client.py

```diff
@@ -0,0 +1,214 @@
+from nisystemlink.clients.result.models import (
+    Result,
+    UpdateResultRequest,
+)
+from uplink import Field, Query, retry, returns
+
+from . import models
+
+
+@retry(when=retry.when.status([429, 503, 504]), stop=retry.stop.after_attempt(5))
+class ResultClient(BaseClient):
+
+    def __init__(self, configuration: Optional[core.HttpConfiguration] = None):
+        """Initialize an instance.
+
+        Args:
+            configuration: Defines the web server to connect to and information about
+                how to connect. If not provided, the
+                :class:`HttpConfigurationManager <nisystemlink.clients.core.HttpConfigurationManager>`
+                is used to obtain the configuration.
+
+        Raises:
+            ApiException: if unable to communicate with the Result Service.
+        """
+        if configuration is None:
+            configuration = core.HttpConfigurationManager.get_configuration()
+        super().__init__(configuration, base_path="/nitestmonitor/v2/")
+
+    @post("results", args=[Field("results")])
+    def create_results(self, results: List[Result]) -> models.ResultsPartialSuccess:
+        """Creates one or more results and returns errors for failed creations.
+
+        Args:
+            results: A list of results to attempt to create.
+
+        Returns: A list of created results, results that failed to create, and errors for
+            failures.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` service of provided invalid
+                arguments.
+        """
+        ...
+
+    @get(
+        "results",
+        args=[Query("continuationToken"), Query("take"), Query("returnCount")],
+    )
+    def get_results_paged(
+        self,
+        continuation_token: Optional[str] = None,
+        take: Optional[int] = None,
+        return_count: Optional[bool] = None,
+    ) -> models.PagedResults:
+        """Reads a list of results.
+
+        Args:
+            continuation_token: The token used to paginate results.
+            take: The number of results to get in this request.
+            return_count: Whether or not to return the total number of results available.
+
+        Returns:
+            A list of results.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @get("results/{id}")
+    def get_result(self, id: str) -> models.Result:
+        """Retrieves a single result by id.
+
+        Args:
+            id (str): Unique ID of a result.
+
+        Returns:
+            The single result matching `id`
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("query-results")
+    def query_results_paged(
```

Comment:
I'd rather this just be query_results. That is inconsistent with products but we'll get products changed to match this

Reply:
Same for get_results_paged -> get_results